Include/internal/pycore_frame.h

```diff
@@ -58,14 +58,20 @@ typedef struct _PyInterpreterFrame {
     PyObject *f_builtins; /* Borrowed reference. Only valid if not on C stack */
     PyObject *f_locals; /* Strong reference, may be NULL. Only valid if not on C stack */
     PyFrameObject *frame_obj; /* Strong reference, may be NULL. Only valid if not on C stack */
-    /* The instruction that is currently executing (possibly not started yet). */
+    /* When a frame is executing, instr_ptr points to the instruction currently executing.
+     * During a call, it points to the call instruction.
+     * In a suspended frame, it points to the instruction that would execute
+     * if the frame were to resume.
+     */
     _Py_CODEUNIT *instr_ptr;
     int stacktop;  /* Offset of TOS from localsplus  */
-    /* The next_instr_offset determines where the next instruction is relative
-     * to instr_ptr. It enables us to keep instr_ptr pointing to the current
-     * instruction until it is time to begin executing the next one, which is
-     * necessary for tracebacks and tracing. */
-    uint16_t next_instr_offset;
+    /* The return_offset determines where a `RETURN` should go in the caller,
+     * relative to `instr_ptr`.
+     * It is only meaningful to the callee,
+     * so it needs to be set in any CALL (to a Python function)
+     * or SEND (to a coroutine or generator).
```

Comment:
Maybe also mention `BINARY_SUBSCR_GETITEM`, `LOAD_ATTR_PROPERTY`, and others? And what about `YIELD_VALUE`? Or `ENTER_EXECUTOR`? Maybe it's better to describe this as some condition or property of instructions (e.g. manipulating frames) rather than naming specific instructions by name?